Zoo/src/ua/lviv/iot/algo/part1/lab1/Zoo.java

```diff
@@ -0,0 +1,80 @@
+package ua.lviv.iot.algo.part1.lab1;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+
+
+public class Zoo {
+    private String name;
+    private String location;
+    private double area;
+    private int capacity;
+    private static Zoo instance;
+    public static Zoo getInstance() {
+        if (instance == null) {
+            return new Zoo();
+        }
+        return instance;
+    }
+    @Override
+    public String toString() {
+        return ("This " + name
+                + " zoo has capacity " + capacity
+                + " animals, has area " + area
+                + " km^2, also it locates in " + location);
+    }
+    public void increaseCapacity(int number) {
+        capacity = capacity + number;
+    }
+    public double splitArea() {
+        if (area == 0) {
+            System.out.println("Nothing to split");
+            return area;
+        }
+        area = area / 2;
+        return area;
+    }
+    public double addNewRegion(double area) {
```

Comment:
This method also does not return a value. All your methods only can do one thing, tht is described in a method name